drivers: spi: Add shim for nrfx SPIS driver

anangl opened the pull request:
This adds a translation layer to make the nrfx driver for the nRF SPIS
(SPI Slave with EasyDMA) peripheral accessible via the Zephyr's API.

Depends on #6985, in particular on the commit bb66f3f3704ed3adcda7e46c8125fed2c54af91f.

anangl committed:
drivers: spi: Add shim for nrfx SPIS driver

This adds a translation layer to make the nrfx driver for the nRF SPIS
(SPI Slave with EasyDMA) peripheral accessible via the Zephyr's API.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl committed:
drivers: spi: Add missing periods in Kconfig.nrfx

Quite a few sentences in help descriptions and comments were not ended
with periods.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

carlescufi commented on ext/hal/nordic/nrfx_glue.h:
```diff
@@ -176,6 +176,37 @@ extern "C" {
+ *
+ * @return Interrupt number associated with the specified peripheral.
+ */
+/* TODO - this macro should become a part of <nrfx_common.h> */
```
Can you add an entry in the nrfx bug tracking system to get rid of this once it's part of nrfx_common.h ?

anangl replied:
[Done](https://github.com/NordicSemiconductor/nrfx/issues/18).

anangl commented:
@tbursztyka I have some doubts regarding `spi_context`:
1. In slave mode it does not automatically release the lock, you need to explicitly call `spi_release` before calling `spi_transceive` again. Is it still intented? The expected behavior is not documented. IIRC it was earlier, but it is no longer after the last update.
2. It returns properly the number of received frames (`ctx->recv_frames`) only in a synchronous operation. In an asynchronous one this value should be passed as the parameter to `k_poll_signal`, because when `spi_transceive_async` returns, it is not known yet.
3. When you provide both master and slave implementations, `ctx->recv_frames` is counted in both modes (even if they are done in separate code, like in the nRF case), and not used then for master. Wouldn't it be better to leave it to particular slave implementations to provide this value in the most convenient way for them? As you can see [here](https://github.com/zephyrproject-rtos/zephyr/pull/7003/commits/cd297fbe6788d1b20f4f209832fc5885c2ab0a9e#diff-4550e1ad9023ccde2007205a5a77b10bR225), in nRF implementation I can get it from a peripheral register after the transfer is done.

galak commented on drivers/spi/spi_nrfx_spi.c:
```diff
@@ -270,24 +270,12 @@ static int init_spi(struct device *dev, const nrfx_spi_config_t *config)
 	static int spi_##idx##_init(struct device *dev)			\
 	{								\
-		IRQ_CONNECT(NRF_SPI_IRQ_NUMBER(idx),			\
+		IRQ_CONNECT(NRFX_IRQ_NUMBER_GET(NRF_SPI##idx),		\
```
Why isn't the IRQ number coming from device tree?

tbursztyka commented:
@anangl 
1) That was the intention indeed, but it's a left-over of the first slave attemps. will get rid of that behavior.
1) yep that's was forgotten
1) why not, but that was just super simple to do it in spi_context.h (and actually, it makes things easy for the previous point)

Will send patches

nashif commented:
is this still DNM?

anangl committed:
boards: 96_carbon_nrf51: Update SPI driver default configuration

Update names of the configuration entries so that the new nRF SPIS
driver can be used on this board.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl commented:
@nashif Now that #6985 is merged, this one is no longer a DNM.

anangl commented:
Added one more commit that updates the default configuration for 96b_carbon_nrf51.

carlescufi commented:
@anangl there's still an unanswered question from @galak about why the IRQ number is not coming from DT. I assume  you will send a PR to switch SPI to DT later?

tbursztyka commented:
@carlescufi I have one PR being prepared for moving all spi (controllers & devices) to DT.

anangl commented:
@carlescufi Yes, that was the plan. And I answered the same question from @galak in #6985. I forgot about the one here.
But now I'll wait for the PR from @tbursztyka.

carlescufi commented:
@anangl ok, so if you fix CI we can merge this then

anangl committed:
tests: net:  Make "app" tests dependent on netif

This will prevent building these tests for boards not supposed to run
a TCP/IP stack, particularly 96b_carbon_nrf51 where test-with-dns
would not fit into 32 kB of RAM.

Signed-off-by: Andrzej Głąbek <andrzej.glabek@nordicsemi.no>

anangl commented:
Added a commit that made CI green. Ready to merge, I guess.
